benchmarks/babelstream/src/benchmarkTest.cpp

```diff
@@ -0,0 +1,434 @@
+    constexpr auto blockThreadExtentMain = 1024;
+
+    //!
+    //! \brief FuzzyEqual compares two floating point or integral type values
+    //!
+    template<typename T>
+    bool FuzzyEqual(T a, T b)
+    {
+        if constexpr(std::is_floating_point_v<T>)
+        {
+            return std::fabs(a - b) < std::numeric_limits<T>::epsilon() * static_cast<T>(1000.0);
+        }
+        else if constexpr(std::is_integral_v<T>)
+        {
+            return a == b;
+        }
+        else
+        {
+            static_assert(
+                std::is_floating_point_v<T> || std::is_integral_v<T>,
+                "FuzzyEqual<T> is only supported for integral or floating-point types.");
+        }
+    }
+
+    std::string getCurrentTimestamp();
+
+    //!
+    //! \brief getCurrentTimestamp gets the time-stamp and returns it as a string
+    //!
+    std::string getCurrentTimestamp()
+    {
+        auto now = std::chrono::system_clock::now();
+        auto now_c = std::chrono::system_clock::to_time_t(now);
+        std::stringstream ss;
+        ss << std::put_time(std::localtime(&now_c), "%Y-%m-%d %X");
+        return ss.str();
+    }
+
+} // namespace
+
+//! Vector summation kernel
+struct AddKernel
+{
+    //! The kernel entry point
+    //! \tparam TAcc The accelerator environment to be executed on.
+    //! \tparam T The data type
+    //! \param acc The accelerator to be executed on.
+    //! \param a Pointer for vector a
+    //! \param b Pointer for vector b
+    //! \param c Pointer for result vector c
+    template<typename TAcc, typename T>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc, T const* a, T const* b, T* c) const
+    {
+        auto const i = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0];
+        c[i] = a[i] + b[i];
+    }
+};
+
+//! Scalat and vector multiplication kernel
+struct MulKernel
+{
+    //! The kernel entry point
+    //! \tparam TAcc The accelerator environment to be executed on.
+    //! \tparam T The data type
+    //! \param acc The accelerator to be executed on.
+    //! \param b Pointer for vector b
+    //! \param c Pointer for result vector c
+    template<typename TAcc, typename T>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc, T* const b, T* c) const
+    {
+        const T scalar = static_cast<T>(startScalar);
+        auto const i = alpaka::getIdx<alpaka::Grid, alpaka::Threads>(acc)[0];
+        c[i] = scalar * b[i];
+    }
+};
+
+//! Vector copying kernel
+struct CopyKernel
+{
+    //! The kernel entry point
+    //! \tparam TAcc The accelerator environment to be executed on.
+    //! \tparam T The data type
+    //! \param acc The accelerator to be executed on.
+    //! \param a Pointer for vector a
+    //! \param c Pointer for result vector c
+    template<typename TAcc, typename T>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc, T const* a, T* c) const
+    {
+        using Dim = alpaka::Dim<TAcc>;
+        if constexpr(Dim::value == 1)
```

Comment:
would be a static_assert() bettert? If the dimension is not 1, the kernel simply do nothing.

Reply:
then wont the test fail because tests are triggered by different types? Let me try.

Reply:
ok, i had set the test type at the test. You are right.